From 27afa5a4ea8de1679603f5871935096280d0b12e Mon Sep 17 00:00:00 2001
From: David Michael <david.michael@coreos.com>
Date: Tue, 13 Jun 2017 13:20:16 -0700
Subject: [PATCH 22/29] sysvinit: Create the socket directory at runtime

This better supports non-systemd configurations with tmpfs on /run.
---
 src/pesign.sysvinit.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/pesign.sysvinit.in b/src/pesign.sysvinit.in
index d8fffca..dc508d8 100644
--- a/src/pesign.sysvinit.in
+++ b/src/pesign.sysvinit.in
@@ -20,6 +20,9 @@ RETVAL=0
 
 start(){
     echo -n "Starting pesign: "
+    mkdir /var/run/pesign 2>/dev/null &&
+        chown pesign:pesign /var/run/pesign &&
+        chmod 0770 /var/run/pesign
     daemon /usr/bin/pesign --daemonize
     RETVAL=$?
     echo
-- 
2.13.4

